### ISSUE TITLE
Issues with compose build

### ISSUE THREAD

**Issue by OmgImAlexis** [razorgirl/nzedb-docker#3]
`tar xf` doesn't seem to work properly but this fixes it.

```
Step 12 : RUN cd /tmp &&   wget http://humdi.net/vnstat/vnstat-1.16.tar.gz &&   tar xf vnstat-1.16.tar.gz &&   cd vnstat-1.16 &&   ./configure &&   make &&   make install
 ---> Running in 0d3a242bdbe4
--2017-01-08 03:38:03--  http://humdi.net/vnstat/vnstat-1.16.tar.gz
Resolving humdi.net... 213.168.251.41, 2a01:7e00::f03c:91ff:fe37:3421
Connecting to humdi.net|213.168.251.41|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 211844 (207K) [application/x-gzip]
Saving to: 'vnstat-1.16.tar.gz'

     0K .......... .......... .......... .......... .......... 24% 79.3K 2s
    50K .......... .......... .......... .......... .......... 48%  157K 1s
   100K .......... .......... .......... .......... .......... 72% 5.31M 0s
   150K .......... .......... .......... .......... .......... 96% 4.96M 0s
   200K ......                                                100% 85.8K=1.0s

2017-01-08 03:38:08 (197 KB/s) - 'vnstat-1.16.tar.gz' saved [211844/211844]

tar: invalid tar magic
ERROR: Service 'nzedb' failed to build: The command '/bin/sh -c cd /tmp &&   wget http://humdi.net/vnstat/vnstat-1.16.tar.gz &&   tar xf vnstat-1.16.tar.gz &&   cd vnstat-1.16 &&   ./configure &&   make &&   make install' returned a non-zero code: 1
```

```diff
diff --git a/nzedb/Dockerfile b/nzedb/Dockerfile
index 57e225e..39d5b08 100644
--- a/nzedb/Dockerfile
+++ b/nzedb/Dockerfile
@@ -35,7 +35,7 @@ RUN apk add htop bwm-ng
 RUN \
   cd /tmp && \
   wget http://humdi.net/vnstat/vnstat-1.16.tar.gz && \
-  tar xf vnstat-1.16.tar.gz && \
+  tar -xzf vnstat-1.16.tar.gz && \
   cd vnstat-1.16 && \
   ./configure && \
   make && \
@@ -46,7 +46,7 @@ RUN \
   apk add perl perl-dbi perl-dbd-mysql perl-term-readkey && \
   cd /tmp && \
   wget http://jeremy.zawodny.com/mysql/mytop/mytop-1.6.tar.gz && \
-  tar xf mytop-1.6.tar.gz && \
+  tar -xzf mytop-1.6.tar.gz && \
   cd mytop-1.6 && \
   perl Makefile.PL && \
   make && \
@@ -56,7 +56,7 @@ RUN \
 RUN \
   cd /tmp && \
   wget https://mediaarea.net/download/binary/mediainfo/0.7.91/MediaInfo_CLI_0.7.91_GNU_FromSource.tar.xz && \
-  tar xf MediaInfo_CLI_0.7.91_GNU_FromSource.tar.xz && \
+  tar -xJf MediaInfo_CLI_0.7.91_GNU_FromSource.tar.xz && \
   cd MediaInfo_CLI_GNU_FromSource && \
   ./CLI_Compile.sh && \
   cd MediaInfo/Project/GNU/CLI && make install
```

**Comment by razorgirl** [comment]
Go ahead and open a pull request then! ;)

**Comment by OmgImAlexis** [comment]
Before I open a PR, do you know why pip3 would be failing?

Maybe pip and pip3 need to be run after they've been installed?

```
Step 16 : RUN apk add python python3 py-pip &&   pip install --upgrade pip &&   pip install cymysql &&   pip install pynntp &&   pip install socketpool &&   pip list &&   pip3 install --upgrade pip &&   pip3 install cymysql &&   pip3 install pynntp &&   pip3 install socketpool &&   pip3 list
 ---> Running in 891fc19eedbc
(1/6) Installing gdbm (1.11-r1)
(2/6) Installing readline (6.3.008-r4)
(3/6) Installing python (2.7.12-r0)
(4/6) Installing py-setuptools (18.8-r0)
(5/6) Installing py-pip (7.1.2-r0)
(6/6) Installing python3 (3.5.1-r0)
Executing busybox-1.24.2-r0.trigger
OK: 397 MiB in 109 packages
Collecting pip
  Downloading pip-9.0.1-py2.py3-none-any.whl (1.3MB)
Installing collected packages: pip
  Found existing installation: pip 7.1.2
    Uninstalling pip-7.1.2:
      Successfully uninstalled pip-7.1.2
Successfully installed pip-9.0.1
Collecting cymysql
  Downloading cymysql-0.8.9.tar.gz
Installing collected packages: cymysql
  Running setup.py install for cymysql: started
    Running setup.py install for cymysql: finished with status 'done'
Successfully installed cymysql-0.8.9
Collecting pynntp
  Downloading pynntp-0.8.4.tar.gz
Collecting dateutils (from pynntp)
  Downloading dateutils-0.6.6.tar.gz
Collecting argparse (from dateutils->pynntp)
  Downloading argparse-1.4.0-py2.py3-none-any.whl
Collecting python-dateutil (from dateutils->pynntp)
  Downloading python_dateutil-2.6.0-py2.py3-none-any.whl (194kB)
Collecting pytz (from dateutils->pynntp)
  Downloading pytz-2016.10-py2.py3-none-any.whl (483kB)
Collecting six>=1.5 (from python-dateutil->dateutils->pynntp)
  Downloading six-1.10.0-py2.py3-none-any.whl
Installing collected packages: argparse, six, python-dateutil, pytz, dateutils, pynntp
  Running setup.py install for dateutils: started
    Running setup.py install for dateutils: finished with status 'done'
  Running setup.py install for pynntp: started
    Running setup.py install for pynntp: finished with status 'done'
Successfully installed argparse-1.4.0 dateutils-0.6.6 pynntp-0.8.4 python-dateutil-2.6.0 pytz-2016.10 six-1.10.0
Collecting socketpool
  Downloading socketpool-0.5.3.tar.gz
Installing collected packages: socketpool
  Running setup.py install for socketpool: started
    Running setup.py install for socketpool: finished with status 'done'
Successfully installed socketpool-0.5.3
DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.
cymysql (0.8.9)
dateutils (0.6.6)
pip (9.0.1)
pynntp (0.8.4)
python-dateutil (2.6.0)
pytz (2016.10)
setuptools (18.8)
six (1.10.0)
socketpool (0.5.3)
/bin/sh: pip3: not found
ERROR: Service 'nzedb' failed to build: The command '/bin/sh -c apk add python python3 py-pip &&   pip install --upgrade pip &&   pip install cymysql &&   pip install pynntp &&   pip install socketpool &&   pip list &&   pip3 install --upgrade pip &&   pip3 install cymysql &&   pip3 install pynntp &&   pip3 install socketpool &&   pip3 list' returned a non-zero code: 127
```

**Comment by OmgImAlexis** [comment]
Okay, `python3 -m ensurepip ` fixes the pip3 issue but now I'm stuck on the PHP modules installing.

Did the build work for you? Seems quite a bit of it is broken.

**Comment by OmgImAlexis** [comment]
Bumping the Alpine version seems to have done the trick with getting PHP 5 installed correctly.

**Comment by razorgirl** [comment]
This is really weird. I tested the build thoroughly, like 20 times. Which OS and Docker version are you building on?

**Comment by OmgImAlexis** [comment]
OSX 10.12.1 (16B2659)
Docker version 1.12.5, build 7392c3b

From the looks of it php-5 isn't even included in alpine 3.3, only 3.4

**Comment by OmgImAlexis** [comment]
```diff
diff --git a/nzedb/Dockerfile b/nzedb/Dockerfile
index 57e225e..a8ee4c8 100644
--- a/nzedb/Dockerfile
+++ b/nzedb/Dockerfile
@@ -3,7 +3,7 @@
 # Create a quick and clean dev environment
 #
 
-FROM alpine:3.3
+FROM alpine:3.4
 MAINTAINER razorgirl <https://github.com/razorgirl>
 
 # Set correct environment variables.
@@ -35,7 +35,7 @@ RUN apk add htop bwm-ng
 RUN \
   cd /tmp && \
   wget http://humdi.net/vnstat/vnstat-1.16.tar.gz && \
-  tar xf vnstat-1.16.tar.gz && \
+  tar -xzf vnstat-1.16.tar.gz && \
   cd vnstat-1.16 && \
   ./configure && \
   make && \
@@ -46,7 +46,7 @@ RUN \
   apk add perl perl-dbi perl-dbd-mysql perl-term-readkey && \
   cd /tmp && \
   wget http://jeremy.zawodny.com/mysql/mytop/mytop-1.6.tar.gz && \
-  tar xf mytop-1.6.tar.gz && \
+  tar -xzf mytop-1.6.tar.gz && \
   cd mytop-1.6 && \
   perl Makefile.PL && \
   make && \
@@ -56,7 +56,7 @@ RUN \
 RUN \
   cd /tmp && \
   wget https://mediaarea.net/download/binary/mediainfo/0.7.91/MediaInfo_CLI_0.7.91_GNU_FromSource.tar.xz && \
-  tar xf MediaInfo_CLI_0.7.91_GNU_FromSource.tar.xz && \
+  tar -xJf MediaInfo_CLI_0.7.91_GNU_FromSource.tar.xz && \
   cd MediaInfo_CLI_GNU_FromSource && \
   ./CLI_Compile.sh && \
   cd MediaInfo/Project/GNU/CLI && make install
@@ -64,14 +64,20 @@ RUN \
 # Install ffmpeg, p7zip, unrar and lame.
 RUN apk add ffmpeg unrar lame p7zip
 
-# Install Python modules.
+# Install Python 2 modules.
 RUN \
-  apk add python python3 py-pip && \
+  apk add python py-pip && \
   pip install --upgrade pip && \
   pip install cymysql && \
   pip install pynntp && \
   pip install socketpool && \
-  pip list && \
+  pip list
+
+# Install Python 3 modules
+RUN apk add python3 && \
+  python3 -m ensurepip && \
+  rm -r /usr/lib/python*/ensurepip && \
+  pip3 install --upgrade pip setuptools && \
   pip3 install --upgrade pip && \
   pip3 install cymysql && \
   pip3 install pynntp && \
```

**Comment by razorgirl** [comment]
Now I see the problem. I built on 3.5. Could you check if it builds correctly as is with 3.5? I have probably overlooked updating the version from 3.3 to 3.5 before committing.

**Comment by OmgImAlexis** [comment]
Will do.

**Comment by razorgirl** [comment]
In case that does not work, just use `alpine` as an image. That's what I started out with. 3.5 should be identical.

**Comment by OmgImAlexis** [comment]
It was working fine in 3.4 but 3.5 throws errors.

```
Step 16 : RUN apk add python python3 py-pip &&   pip install --upgrade pip &&   pip install cymysql &&   pip install pynntp &&   pip install socketpool &&   pip list &&   pip3 install --upgrade pip &&   pip3 install cymysql &&   pip3 install pynntp &&   pip3 install socketpool &&   pip3 list
 ---> Running in d460b5f12642
ERROR: unsatisfiable constraints:
  py-pip (virtual):
    provided by: py2-pip
    required by: world[py-pip]
ERROR: Service 'nzedb' failed to build: The command '/bin/sh -c apk add python python3 py-pip &&   pip install --upgrade pip &&   pip install cymysql &&   pip install pynntp &&   pip install socketpool &&   pip list &&   pip3 install --upgrade pip &&   pip3 install cymysql &&   pip3 install pynntp &&   pip3 install socketpool &&   pip3 list' returned a non-zero code: 1
```

**Comment by OmgImAlexis** [comment]
Just using `apline`causes the same error as above.

**Comment by razorgirl** [comment]
Fixed by PR #5
